### PR TITLE
Eric/app 745 handle custom feed failure cases

### PR DESCRIPTION
**Pull request by estrattonbailey**
Adds an "inner" header to custom feeds that showcases the author and provides a context menu in places there wasn't one before. Also revises how errors are rendered in these cases.

https://github.com/bluesky-social/social-app/assets/4732330/2b803a74-5da9-47d1-a903-3f6e6591f8b8

Screenshot 2023-10-31 at 1 50 38 PM
Screenshot 2023-10-31 at 1 56 25 PM
Screenshot 2023-10-31 at 1 56 41 PM
Screenshot 2023-10-31 at 1 57 47 PM

### CODE REVIEW

File: src/lib/hooks/useCustomFeed.ts [hunk 1]
```diff
@@ -7,11 +7,16 @@ export function useCustomFeed(uri: string): CustomFeedModel | undefined {
...
-      })
-      const view = res.data.view
-      return view
+      try {
+        const res = await store.agent.app.bsky.feed.getFeedGenerator({
+          feed: uri,
+        })
+        const view = res.data.view
+        return view
+      } catch (e) {
```
Comment (estrattonbailey) [inline src/lib/hooks/useCustomFeed.ts:16]: This was throwing an uncaught error

File: src/lib/strings/errors.ts [hunk 2]
```diff
@@ -22,6 +22,7 @@ export function isNetworkError(e: unknown) {
   return (
     str.includes('Abort') ||
     str.includes('Network request failed') ||
-    str.includes('Failed to fetch')
+    str.includes('Failed to fetch') ||
+    str.includes('NetworkError when attempting to fetch resource')
```
Comment (estrattonbailey) [inline src/lib/strings/errors.ts:26]: Just want to catch this case too

File: src/state/models/feeds/posts.ts [hunk 3]
```diff
@@ -47,6 +52,7 @@ export class PostsFeedModel {
   isBlockedBy = false
   error = ''
   loadMoreError = ''
+  cleanError?: PostsFeedModelError = undefined
```
Comment (estrattonbailey) [inline src/state/models/feeds/posts.ts:55]: One idea for attaching more info to the error so we know more about what happened

File: src/view/com/feeds/CustomFeedContextMenu.tsx
Comment (estrattonbailey) [inline src/view/com/feeds/CustomFeedContextMenu.tsx:1]: This whole file is lifted from `CustomFeed` screen, no functional changes.

File: src/view/com/posts/Feed.tsx [hunk 4]
```diff
@@ -51,16 +64,23 @@ export const Feed = observer(function Feed({
...
 
   const data = React.useMemo(() => {
     let feedItems: any[] = []
     if (feed.hasLoaded) {
-      if (feed.hasError) {
+      if (isCustomFeed) {
+        feedItems.push(FEED_INNER_HEADER)
+      }
+      if (feed.hasError && !isCustomFeed) {
+        // applies to our internal algo feeds only
```
Comment (estrattonbailey) [inline src/view/com/posts/Feed.tsx:83]: If it's a custom feed, the error message will show within the "inner" header component, which this PR introduces. If it's not a custom feed, we show a similar looking error message outside the header as we have been.

File: src/view/com/posts/Feed.tsx [hunk 5]
```diff
@@ -115,15 +136,20 @@ export const Feed = observer(function Feed({
...
-      } else if (item === ERROR_ITEM) {
+      if (item === FEED_INNER_HEADER_LOADING) {
+        return <FeedHeaderLoading />
+      } else if (item === FEED_INNER_HEADER) {
         return (
-          <ErrorMessage
-            message={feed.error}
-            onPressTryAgain={onPressTryAgain}
+          <FeedInnerHeader
+            showContextMenu={showFeedHeaderContextMenu}
```
Comment (estrattonbailey) [inline src/view/com/posts/Feed.tsx:144]: This prop is for use on the `CustomFeed` screen, which already uses this context menu.

File: src/view/com/posts/Feed.tsx [hunk 6]
```diff
@@ -158,7 +191,13 @@ export const Feed = observer(function Feed({
       <FlatList
         testID={testID ? `${testID}-flatlist` : undefined}
         ref={scrollElRef}
-        data={!feed.hasLoaded ? [LOADING_ITEM] : data}
+        data={
+          !feed.hasLoaded
+            ? isCustomFeed
+              ? [FEED_INNER_HEADER_LOADING, LOADING_ITEM]
+              : [LOADING_ITEM]
```
Comment (estrattonbailey) [inline src/view/com/posts/Feed.tsx:198]: If custom feed, show an additional skeleton UI component for the inner header

File: src/view/com/posts/Feed.tsx [hunk 7]
```diff
@@ -189,6 +228,150 @@ export const Feed = observer(function Feed({
...
+              }}>
+              <FontAwesomeIcon
+                icon="ellipsis"
+                size={18}
+                color={pal.colors.textLight}
+              />
+            </View>
+          </CustomFeedContextMenu>
+        ) : (
+          <View style={{height: 24}} />
```
Comment (estrattonbailey) [inline src/view/com/posts/Feed.tsx:342]: Here to prevent layout shift when loaded

**Comment by pfrazee** [comment]
I like the premise of what you’re going for with the bottom bar, but I think we need a design in the tabs bar that specifically accommodates this

**Comment by pfrazee** [comment]
Superseded by https://github.com/bluesky-social/social-app/pull/1798

**Pull request closed without merging**